module-system/sov-modules-api/src/lib.rs

```diff
@@ -221,9 +223,26 @@ pub trait ModuleInfo {
     fn address(&self) -> &<Self::Context as Spec>::Address;
 }
 
-// /// this is the struct that users need to instantiate
-// /// with a storage object to convert into an RPC
-// #[derive(Clone)]
-// pub struct RpcStorage<C: Context> {
-//     pub storage: C::Storage,
-// }
+pub trait StateTransitionRunner<T: StateTransitionConfig, Vm: Zkvm> {
+    /// The parameters of the state transition function which are set at runtime. For example,
+    /// the runtime config might contain path to a data directory.
+    type RuntimeConfig;
+    type Inner: StateTransitionFunction<Vm>;
+    type Context: Context;
+    // TODO: decide if `new` also requires <Self as StateTransitionFunction>::ChainParams as an argument
+    /// Create a state transition runner
+    fn new(runtime_config: Self::RuntimeConfig) -> Self;
+
+    /// Return a reference to the inner STF implementation
+    fn inner(&self) -> &Self::Inner;
+
+    /// Return a mutable reference to the inner STF implementation
+    fn inner_mut(&mut self) -> &mut Self::Inner;
+
+    /// Return a mutable reference to the inner STF implementation
+    fn get_storage(&self) -> <Self::Context as Spec>::Storage;
+
+    // /// Report if the state transition function has been initialized.
+    // /// If not, node implementations should respond by running `init_chain`
+    // fn has_been_initialized(&self) -> bool;
+}
```

Comment:
Hmm, thinking about this change in more depth,  I think it becomes impossible to run the full node without taking a dependency on sov-modules-api (you need the state-transition runner to get started, right?). 

If so, I guess we need to go the route of splitting this into two traits: the old StateTransitionRunner (which would live in the core/rollup-interface crate) and a new trait like RpcRunner which lives in sov-modules-api and has the `Context` type and `get_storage` method. That way, users can still run the default full-node implementation even if they don't use sov-modules, but they get an easy way to add RPC by just importing the trait and running an initializer if they do use sov-modules

Reply:
yeah that makes sense. was thinking of a way to avoid another trait implementation that they needed to provide, but being explicit is better than tying them together